### PR TITLE
[#645] Add a tutorial for external mqtt clients

### PR DESCRIPTION
**Pull request by brusdev**

### CODE REVIEW

File: docs/tutorials/external_mqtt_clients.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+title: "Connecting to the broker from external mqtt clients"  
+description: "Espose an mqtt acceptor with ssl enabled to accept connections from external mqtt clients"
```
Comment (michalxo) [inline docs/tutorials/external_mqtt_clients.md:3]: Expose

Comment (brusdev) [reply docs/tutorials/external_mqtt_clients.md]: good catch!

File: docs/tutorials/external_mqtt_clients.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+draft: false
+images: []
+menu:
+  docs:
+    parent: "tutorials"
+weight: 110
+toc: true
+---
+
+When you expose an acceptor to external clients (that is, by setting the value of the expose parameter to true), the Operator automatically creates a an ingress on Kubernetes or a route on OpenShift for each broker pod of the deployment. An external client can connect to the broker by specifying the full host name of the ingress/route created for the broker pod.
```
Comment (michalxo) [inline docs/tutorials/external_mqtt_clients.md:13]: a an ingress

File: docs/tutorials/external_mqtt_clients.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+# Prerequisite
+Before you start you need have access to a running Kubernetes cluster environment. A [Minikube](https://minikube.sigs.k8s.io/docs/start/) with [Ingress](https://kubernetes.io/docs/tasks/access-application-cluster/ingress-minikube/) running on your laptop will just do fine. The ArtemisCloud operator also runs in a Openshift cluster environment like [CodeReady Container](https://developers.redhat.com/products/codeready-containers/overview). In this blog we assume you have Kubernetes cluster environment. Execute the following command to enable Ingress in minikube:
```
Comment (michalxo) [inline docs/tutorials/external_mqtt_clients.md:16]: you need TO have access

Comment (michalxo) [inline docs/tutorials/external_mqtt_clients.md:16]:
```suggestion
Before you start you need to have access to a running Kubernetes cluster environment. A [Minikube](https://minikube.sigs.k8s.io/docs/start/) with [Ingress](https://kubernetes.io/docs/tasks/access-application-cluster/ingress-minikube/) running on your laptop will just do fine. The ArtemisCloud operator also runs in Openshift cluster environment like [CodeReady Container](https://developers.redhat.com/products/codeready-containers/overview). In this blog we assume you have Kubernetes cluster environment. Execute the following command to enable Ingress in minikube:
```

File: docs/tutorials/external_mqtt_clients.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      protocols: mqtt
+      sslEnabled: true
+      sslSecret: my-tls-secret
+  env:
+    - name: JAVA_ARGS_APPEND
+      value: -Djavax.net.debug=all
+EOF
+```
+
+### Publish a message with mosquitto_pub
```
Comment (michalxo) [inline docs/tutorials/external_mqtt_clients.md:71]: Wouldn't be helpful to provide some get/download link to image/mosquito_pub app? As it's tutorial, I am stupid and can't use google properly. Would prefer to just click on some useful link. :-)

https://mosquitto.org/download/ ?

**Comment by michalxo** [comment]
Provided few comments. Else LGTM. I will follow this guide in next days as well.

File: docs/tutorials/external_mqtt_clients.md (hunk 2 above)
Comment (brusdev) [reply docs/tutorials/external_mqtt_clients.md]: fixed

File: docs/tutorials/external_mqtt_clients.md (hunk 3 above)
Comment (brusdev) [reply docs/tutorials/external_mqtt_clients.md]: fixed

File: docs/tutorials/external_mqtt_clients.md (hunk 4 above)
Comment (brusdev) [reply docs/tutorials/external_mqtt_clients.md]: Done, I also added the command to publish a message with mosquitto_pub from the eclipse-mosquitto container.

**Review by michalxo — APPROVED** [review]
LGTM

**Comment by brusdev** [comment]
@michalxo thanks for your review

**Pull request merged**